frontend/src/credit_transfers/components/CreditTransferTextRepresentation.js

```diff
@@ -0,0 +1,145 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import numeral from 'numeral';
+
+import * as NumberFormat from '../../constants/numeralFormats';
+import { CREDIT_TRANSFER_STATUS, CREDIT_TRANSFER_TYPES } from '../../constants/values';
+
+class CreditTransferTextRepresentation extends Component {
+  constructor (props) {
+    super(props);
+
+    this.creditsFrom = this.props.creditsFrom.name;
+    this.creditsTo = this.props.creditsTo.name;
+    this.numberOfCredits = numeral(this.props.numberOfCredits).format(NumberFormat.INT);
+    this.totalValue = numeral(this.props.totalValue).format(NumberFormat.CURRENCY);
+    this.tradeEffectiveDate = (this.props.tradeEffectiveDate)
+      ? moment(this.props.tradeEffectiveDate).format('LL') : 'approval';
+
+    this.tradeStatus = (this.props.status.id === CREDIT_TRANSFER_STATUS.completed.id)
+      ? CREDIT_TRANSFER_STATUS.approved.description : this.props.status.status;
+  }
+
+  _buyAction () {
+    if (this.props.status.id === CREDIT_TRANSFER_STATUS.completed.id) {
```

Comment:
Just a personal preference note: I would do:
`return this.props.status.id === CREDIT_TRANSFER_STATUS.completed.id ? ' bought ' : ' is proposing to buy '`

I think it's easier to read, and your function only has one exit point.

Reply:
sounds good. i have no problems with ternary functions.